storage/filesystem/littlefsv2/source/LittleFileSystem2.cpp

```diff
@@ -241,7 +242,8 @@ int LittleFileSystem2::format(BlockDevice *bd,
     _config.sync            = lfs2_bd_sync;
     _config.read_size       = bd->get_read_size();
     _config.prog_size       = bd->get_program_size();
-    _config.block_size      = lfs2_max(block_size, (lfs2_size_t)bd->get_erase_size());
+    _config.block_size      = lfs2_max(block_size, lfs2_max((lfs2_size_t)bd->get_erase_size(0),
+                                                    (lfs2_size_t)bd->get_erase_size(bd->size())));
```

Comment:
The underlying block device (`bd`) has to be aligned already. But even then, we still need to check the maximum erase size. For example, 128KB + 128KB + 256 KB x N is aligned but `bd->get_erase_size()` (no parameter) returns 128KB which is the size at the base address. Formatting with 128KB will fail with invalid parameter.

(Note: `BlockDevice::erase(address, size)` permits `size` to be multiples of the underlying erase size, so erasing 256KB on 128KB x 2 is no problem as long as things are aligned.)